### ISSUE TITLE
as.section should stop worrying about `waterDepth`

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#1797]
Following from Dan's comment/proposal in #1781 here: https://github.com/dankelley/oce/issues/1781#issuecomment-778765465

> I think as.section() ought to stop worrying about waterDepth if it's not defined in the objects presented to it. Sometimes, as with Argo floats, we just don't know.
> Well, actually, a user could look it up in a bathymetry file, if they really cared, and if they trusted the position recorded for the Argo float. Personally, I'd be a "no" on both. I'd just as soon see plots (which is the main purpose here, I imagine) stop at 1km or 2km or whatever it is. Otherwise, in the average case, half or more of the plot would be gray colour for the bottom. Also, I find that positions are bad in something like 10% of the (high-latitude) data I've been looking at lately.

**Comment by dankelley** [comment]
@richardsc I plan to leave this to after the CRAN submission.  If you think it is simple enough to fix, and important enough, please add the "CRITICAL" label and provide a reprex, and then I can take a look (or you can).

I have some other things tomorrow, so if we want to submit on Friday, we just have today to fix this.

My most pressing thing right now is #1798 because I think that will yield CRAN rejection.

**Comment by richardsc** [comment]
This is not critical, and can wait for the next release.

**Comment by dankelley** [comment]
I think this is fixed in "develop" commit 1e6a08b2ff66da76e153e280cbb17bcd36cb65e3, as shown in the following reprex.

@richardsc do you think this demonstrates that things are OK now?  I didn't do terribly extensive tests, so I'd appreciate it if you could think of other tests to do.

<details><summary>reprex</summary>

``` r
# Top: section with CTDs containing waterDepth, so bottom is shown in grey.
# Bottom panel: a section made from argo data, so bottom is shown.
library(oce)
#> Loading required package: gsw
#> Loading required package: testthat
#> Loading required package: sf
#> Linking to GEOS 3.8.1, GDAL 3.1.4, PROJ 6.3.1
data(section)
library(argoFloats)
a <- readProfiles(getProfiles(subset(subset(getIndex(age=10), ID=6901191), cycle=1:10)))
#> Kept 176 cycles (0.00724%)
#> Kept 10 cycles (5.68%)
sec <- as.section(a[["argos"]])
summary(sec)
#> Section Summary
#> ---------------
#> 
#> * Source: ""
#> * ID:     ""
#> * Overview of stations
#> Index    ID      Lon      Lat  Levels Depth
#>     1       -20.0733  59.5992    1003    NA
#>     2       -18.7849  59.4765    1003    NA
#>     3       -19.3513  60.1866    1003    NA
#>     4       -19.2923  60.4727    1004    NA
#>     5       -17.2529  60.6655    1004    NA
#>     6       -17.9755  61.3591    1002    NA
#>     7       -19.4469  60.7542    1003    NA
#>     8       -20.0200  60.5276    1004    NA
#>     9       -21.0042  60.7458    1002    NA
#>    10       -21.0701  60.5646    1002    NA
#> * Data-quality flags
#>     pressure:            "1" 10029, "4" 1
#>     pressureAdjusted:    "1" 10029, "4" 1
#>     temperature:         "1" 10030
#>     temperatureAdjusted: "1" 10030
#>     salinity:            "1" 10020, "4" 10
#>     salinityAdjusted:    "1" 10020, "4" 10
#> * Processing Log
#> 
#>     - 2021-04-01 14:40:36 UTC: `create 'section' object`
#>     - 2021-04-01 14:40:36 UTC: `as.section(salinity = a[["argos"]])`
par(mfrow=c(2, 1))
plot(section, which="temperature", ylim=c(5000, 0), showBottom=TRUE)
plot(sec, which="temperature", ylim=c(3000, 0), showBottom=TRUE)
```

![](https://i.imgur.com/IXYCyaT.png)

<sup>Created on 2021-04-01 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by dankelley** [comment]
I made up another test, this time where the first argument is a list of CTDs.

<details><summary>reprex</summary>

``` r
# Top: section with CTDs containing waterDepth, so bottom is shown in grey.
# Bottom panel: resulf from as.section() with first 50 stations.
library(oce)
#> Loading required package: gsw
#> Loading required package: testthat
#> Loading required package: sf
#> Linking to GEOS 3.8.1, GDAL 3.1.4, PROJ 6.3.1
data(section)
par(mfrow=c(2, 1))
plot(section, which="temperature", ylim=c(5000, 0), showBottom=TRUE)
ctds <- section[["station"]][1:50]
sec2 <- as.section(ctds)
plot(sec2, which="temperature", ylim=c(5000, 0), showBottom=TRUE)
```

![](https://i.imgur.com/HAXAN07.png)

<sup>Created on 2021-04-01 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>
